src/nwbinspector/checks/time_series.py

```diff
@@ -129,3 +129,14 @@ def check_resolution(time_series: TimeSeries):
         return InspectorMessage(
             message=f"'resolution' should use -1.0 or NaN for unknown instead of {time_series.resolution}."
         )
+
+
+@register_check(importance=Importance.BEST_PRACTICE_VIOLATION, neurodata_type=TimeSeries)
+def check_rate_is_not_zero(time_series: TimeSeries):
```

Comment:
Hm...

Big question is, should this be a VIOLATION or CRITICAL, since any time series with more than one data element on the first axis and a rate of zero is practically invalid?